temporal-serviceclient/src/main/java/io/temporal/serviceclient/WorkflowServiceStubsOptions.java

```diff
@@ -324,13 +336,27 @@ public Builder setRpcTimeout(Duration timeout) {
     /**
      * Sets the rpc timeout value for the following long poll based operations:
      * PollWorkflowTaskQueue, PollActivityTaskQueue, GetWorkflowExecutionHistory. Should never be
-     * below 60000 as this is server side timeout for the long poll. Default is 61000.
+     * below 60 seconds as this is server side timeout for the long poll. Default is 121 seconds.
      */
     public Builder setRpcLongPollTimeout(Duration timeout) {
       this.rpcLongPollTimeout = Objects.requireNonNull(timeout);
       return this;
     }
 
+    /** Sets the rpc timeout for queries. Defaults to 10 seconds. */
+    public void setRpcQueryTimeout(Duration rpcQueryTimeout) {
+      this.rpcQueryTimeout = rpcQueryTimeout;
+    }
+
+    /**
+     * Allows customization of retry options for the outgoing RPC calls to temporal service. Note
+     * that default values should be reasonable for most users, be cautious when changing these
+     * values as it may result in non-trivial issues.
```

Comment:
An example of what those issues could be might be illuminating.

Reply:
Should we say "increased load to the temporal server or insignificant retries"?

Reply:
Done